eng/Versions.props

```diff
@@ -78,6 +78,7 @@
     <MicrosoftVisualStudioPackagesVersion>17.0.487</MicrosoftVisualStudioPackagesVersion>
     <RoslynPackageVersion>4.2.0-2.22108.10</RoslynPackageVersion>
     <VisualStudioLanguageServerProtocolVersion>17.1.11</VisualStudioLanguageServerProtocolVersion>
+    <MicrosoftNetCompilersToolsetVersion>4.2.0-1.final</MicrosoftNetCompilersToolsetVersion>
```

Comment:
Why did this end up being needed?

Reply:
Normally the compiler version (not language version) comes from the SDK, so we would have to wait to use a version of the .NET 7 SDK that ships with this compiler in order to use it. This overrides the SDK, and just tells everything to use the version of the compiler we specify here.